airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/jinja.py

```diff
@@ -74,20 +82,29 @@ def eval(self, input_str: str, config: Config, default: Optional[str] = None, **
             if isinstance(input_str, str):
                 result = self._eval(input_str, context)
                 if result:
-                    return self._literal_eval(result)
+                    return self._literal_eval(result, valid_types)
             else:
                 # If input is not a string, return it as is
                 raise Exception(f"Expected a string. got {input_str}")
         except UndefinedError:
             pass
         # If result is empty or resulted in an undefined error, evaluate and return the default string
-        return self._literal_eval(self._eval(default, context))
+        return self._literal_eval(self._eval(default, context), valid_types)
 
-    def _literal_eval(self, result):
+    def _literal_eval(self, result, valid_types: Optional[Tuple[Type]]):
         try:
-            return ast.literal_eval(result)
+            evaluated = ast.literal_eval(result)
         except (ValueError, SyntaxError):
             return result
+        if not valid_types or (valid_types and isinstance(evaluated, tuple(valid_types))):
+            return evaluated
+        elif valid_types and isinstance(evaluated, dict):
```

Comment:
I'm not sure I get this case. Let's say that as a user, I type `{ "key" : "value" }` as a param and for whatever reasons, the spaces are important. Wouldn't they be removed by this case? I'm wondering if we should only do `elif `isinstance(result, valid_types): return result`. Are there other cases I'm missing?

Reply:
Yep good point, removed the separators.

Reply:
Let's assume the string the user set up as a query param value is `{'updatedDateFrom': '2023-08-20T00:00:00Z'}`, are we comfortable with `{"updatedDateFrom": "2023-08-20T00:00:00Z"}` being sent instead? It feels odd to me that there would be no work-around for the user to send single quotes as we parse it has json

Reply:
Update: talked about this synchronously. We're just going to return `result` if the value is a dictionary instead of casting or dumping to a json string.